spark_client/utils.py

```diff
@@ -259,3 +274,49 @@ def add_deploy_arguments(parser):
         help="Deployment mode for job submission. Default is 'client'.",
     )
     return parser
+
+
+def get_driver_host_conf() -> Any:
+    args_including_conf, remaining_args = parse_arguments_with(
+        [add_logging_arguments, custom_parser, add_conf_arguments]
+    )
+
+    if args_including_conf.conf:
+        conf_args = dict(
+            conf_entry.split("=", 1) for conf_entry in args_including_conf.conf
+        )
+    else:
+        conf_args = {}
+
+    if "spark.driver.host" not in conf_args:
+        spark_driver_host = detect_host()
+        if spark_driver_host is None:
+            raise ValueError(
+                "Please provide spark.driver.host spark configuration option to proceed....."
+            )
+        return spark_driver_host
+    else:
+        return None
+
+
+def detect_host() -> Any:
+    try:
+        host_py = socket.gethostbyname(socket.gethostname()).strip()
+    except Exception:
+        return None
+
+    try:
+        host_bash = (
+            subprocess.check_output(
+                "hostname -I | cut -d' ' -f1", shell=True, stderr=None
```

Comment:
Hi @grobbie - this is where we are detecting the host. By 2 methods, which if give same ip, we use it.

Reply:
sure, but `hostname -f` will give you the fully qualified domain name of the computer. The executors would then use DNS to find the correct IP of the computer hosting the Spark driver. And if DNS doesn't work properly, there are going to be bigger problems. I don't think we should try to guess the IP, especially if the computer is running Docker or LXD, or his multihomed with several NICs (eg. fitted with iLO and public/private IPs).

Reply:
Thanks Rob, that makes sense. Let me verify this.

Reply:
getting ```java.net.UnknownHostException``` in executors when passing the ```hostname -f``` output.

Reply:
I mean, the idea here is not to work in 100% cases, but to work in most of them, and get in most of the cases a cleaner UX (like showed in the tests, [here](https://github.com/canonical/spark-client-snap/pull/36/files#diff-c841d3a6788870a95eca90d4ff1523ea3a3f3a333f7b883b0e78f7b68cd19526R72)). When the auto-inference does not work, the user can of course specify the driver.host at different level of out hierarchical configuratoin system:
* in the env file (which should be static in the machine, there may be a sensible place to specify machine IP)
* in the property file provided via CLI
* as --conf in the CLI

Of course, the logic behind auto-inference can be changed and improved as we go, but this PR should nevertheless pose the structure to allow this